Sources/NTPClient.swift

```diff
@@ -76,106 +75,113 @@ final class NTPClient {
             }
 
-            completion(PDU: PDU.isValidResponse() ? PDU : nil)
+            completion(PDU.isValidResponse() ? PDU : nil)
         }
 
-        let callback = unsafeBitCast(bridgeCallback, AnyObject.self)
+        let callback = unsafeBitCast(bridgeCallback, to: AnyObject.self)
         let retainedCallback = Unmanaged.passRetained(callback)
         let sourceAndSocket = self.sendAsyncUDPQuery(
             to: ip, port: port, timeout: timeout,
-            completion: UnsafeMutablePointer<Void>(retainedCallback.toOpaque())
+            completion: UnsafeMutableRawPointer(retainedCallback.toOpaque())
         )
 
-        timer = BlockTimer.scheduledTimerWithTimeInterval(timeout) { _ in
-            bridgeCallback(nil, NSTimeInterval.infinity)
+        timer = BlockTimer.scheduledTimer(withTimeInterval: timeout, repeated: true) { _ in
+            bridgeCallback(nil, TimeInterval.infinity)
             retainedCallback.release()
 
             if let (source, socket) = sourceAndSocket {
                 CFSocketInvalidate(socket)
-                CFRunLoopRemoveSource(CFRunLoopGetMain(), source, kCFRunLoopCommonModes)
+                CFRunLoopRemoveSource(CFRunLoopGetMain(), source, CFRunLoopMode.commonModes)
             }
         }
     }
 
     // MARK: - Private helpers (NTP Calculation)
 
-    private func offsetFromResponses(responses: [[NTPPacket]]) throws -> NSTimeInterval {
+    private func offset(from responses: [[NTPPacket]]) throws -> TimeInterval {
         let now = currentTime()
         var bestResponses: [NTPPacket] = []
         for serverResponses in responses {
             let filtered = serverResponses
                 .filter { abs($0.originTime - now) < kMaximumResultDispersion }
-                .minElement { $0.delay < $1.delay }
+                .min { $0.delay < $1.delay }
 
             if let filtered = filtered {
                 bestResponses.append(filtered)
             }
         }
 
         if bestResponses.count == 0 {
-            throw NTPNetworkError.NoValidNTPPacketFound
+            throw NTPNetworkError.noValidNTPPacketFound
         }
 
-        bestResponses.sortInPlace { $0.offset < $1.offset }
+        bestResponses.sort { $0.offset < $1.offset }
         return bestResponses[bestResponses.count / 2].offset
     }
 
     // MARK: - Private helpers (CFSocket)
 
-    private func sendAsyncUDPQuery(to ip: InternetAddress, port: Int, timeout: NSTimeInterval,
-                                      completion: UnsafeMutablePointer<Void>) -> (CFRunLoopSource, CFSocket)?
+    private func sendAsyncUDPQuery(to ip: InternetAddress, port: Int, timeout: TimeInterval,
+                                      completion: UnsafeMutableRawPointer) -> (CFRunLoopSource, CFSocket)?
     {
         let callback: CFSocketCallBack = { socket, callbackType, address, data, info in
-            if callbackType == .WriteCallBack {
+            guard let info = info else {
```

Comment:
this `guard` should be after the next `if` and should also call `completion(nil, -1)`